client/src/services/apiProxy.js

```diff
@@ -215,8 +216,10 @@ export async function streamResults(url, cb, errCb) {
     }
 
     function push() {
-        const values = Object.values(results);
-        cb(values);
+        return _.debounce(() => {
```

Comment:
Same thing as above applies here as well. I think around line number 145 we'll want something like...

```
const debouncedCallback = _.debounce(() => {
    const values = Object.values(results);
    cb(values);
}, 250, {leading: true});
```

Then the `push` function just calls that function like...
```
function push() {
    debouncedCallback();
}
```

(or even better you could just refactor away the `push` function all together then)